subsys/dfu/img_util/flash_img.c

```diff
@@ -18,6 +18,19 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #include <dfu/flash_img.h>
 #include <inttypes.h>
 
+#if USE_PARTITION_MANAGER
+#include <pm_config.h>
+#define FLASH_AREA_IMAGE_SECONDARY  PM_MCUBOOT_PARTITIONS_SECONDARY_ID
+#else
+#include <generated_dts_board.h>
+/* DT_FLASH_AREA_IMAGE_XX_YY values used below are auto-generated by DT */
+#ifdef CONFIG_TRUSTED_EXECUTION_NONSECURE
```

Comment:
IMO we should not support non-secure building of the bootloader, so this check is superfluous.
Only the secure version should be used.